address.js

```diff
@@ -0,0 +1,263 @@
+const {validateBuffer, validateDerivationScheme, validateArray} = require("./utils/validation")
+const {chacha20poly1305Encrypt, chacha20poly1305Decrypt, blake2b, sha3_256} = require("./crypto-primitives")
+
+const cbor = require('borc')
+const crc32 = require('./utils/crc32')
+const base58 = require('./utils/base58')
+const CborIndefiniteLengthArray = require('./utils/CborIndefiniteLengthArray')
+
+const AddressTypes = {
+  'BASE': 0b0000,
+  'POINTER': 0b0100,
+  'ENTERPRISE': 0b0110,
+  'BOOTSTRAP': 0b1000,
+  'REWARDS': 0b1110
+}
+
+function validateAddressType(input, addressTypeId) {
+  if (!['BASE', 'POINTER', 'ENTERPRISE', 'BOOTSTRAP', 'REWARDS'].includes(addressTypeId)) {
+    throw new Error("Invalid address type to validate!")
+  }
+  const addressType = {
+    BASE: {
+      BASE_ADDRESS_KEY_KEY: 0,
+      BASE_ADDRESS_SCRIPT_KEY: 1,
```

Comment:
script addresses won't be supported in this version of shelley, let's remove them for now completely so it does not confuse us